### PR TITLE
Workarounds for intel 2021.7

### PR DESCRIPTION
**Pull request by tclune**

### CODE REVIEW

File: src/Nodes/Nodes.F90 [hunk 1]
```diff
@@ -27,6 +27,8 @@ subroutine assign_to_logical(flag, this)
       logical, intent(out) :: flag
       class(YAML_Node), intent(in) :: this
 
+      logical, pointer :: ptr
+
```
Comment (tclune) [inline src/Nodes/Nodes.F90:31]:
```suggestion
```

File: src/Parser.F90 [hunk 2]
```diff
@@ -94,12 +95,16 @@ function new_Parser_schema_name(schema_name,unusable,rc) result(p)
       class(KeywordEnforcer), optional, intent(in) :: unusable
       integer, optional, intent(out) :: rc
 
+      print*,__FILE__,__LINE__
```
Comment (tclune) [inline src/Parser.F90:98]:
```suggestion
```

File: src/Parser.F90 [hunk 3]
```diff
@@ -96,8 +97,10 @@ function new_Parser_schema_name(schema_name,unusable,rc) result(p)
 
       select case (schema_name)
       case ('json','JSON')
+      print*,__FILE__,__LINE__
```
Comment (tclune) [inline src/Parser.F90:100]:
```suggestion
```

File: Examples/Trivial/trivial.F90 [hunk 4]
```diff
@@ -3,11 +3,12 @@ program main
    implicit none
    
    type(Parser) :: p
-   class(AbstractNode), allocatable :: node
+   class(YAML_Node), allocatable :: node
    integer :: prime
    integer :: status
    
-   p = Parser('core')
+!!$   p = Parser('core')
```
Comment (tclune) [inline Examples/Trivial/trivial.F90:10]:
```suggestion
```

File: Examples/Iterators/iterator.F90 [hunk 5]
```diff
@@ -9,7 +9,7 @@ program main
    implicit none
 
    type(Parser) :: p
-   class(AbstractNode), allocatable :: node
+   class(YAML_Node), allocatable :: node
    integer :: status
 
    p = Parser('core')
```
Comment (mathomp4) [inline Examples/Iterators/iterator.F90:15]: Should this be:

```suggestion
   p = Parser()
```

**Pull request merged**